src/utils/window-console.js

```diff
@@ -210,6 +210,61 @@ export function addDataToWindowObject(
     URL.revokeObjectURL(blobUrl);
   };
 
+  // This function extracts MOZ_LOGs saved as markers in a Firefox profile,
+  // using the MOZ_LOG canonical format. All logs are saved as a debug log
+  // because the log level information isn't saved in these markers.
+  target.extractGeckoLogs = function () {
+    function pad(p, c) {
+      return String(p).padStart(c, '0');
+    }
+
+    // This transforms a timestamp to a string as output by mozlog usually.
+    function d2s(ts) {
+      const d = new Date(ts);
+      // new Date rounds down the timestamp (in milliseconds) to the lower integer,
+      // let's get the microseconds and nanoseconds differently.
+      // This will be imperfect because of float rounding errors but still better
+      // than not having them.
+      const ns = Math.trunc((ts - Math.trunc(ts)) * 10 ** 6);
+      return `${d.getFullYear()}-${pad(d.getUTCMonth() + 1, 2)}-${pad(d.getUTCDate(), 2)} ${pad(d.getUTCHours(), 2)}:${pad(d.getUTCMinutes(), 2)}:${pad(d.getUTCSeconds(), 2)}.${pad(d.getUTCMilliseconds(), 3)}${pad(ns, 6)} UTC`;
+    }
+
+    const all = [];
+
+    // This algorithm loops over the raw marker table instead of using the
+    // selectors so that the full marker list isn't generated for all the
+    // threads in the profile.
+    const profile = selectorsForConsole.profile.getProfile(getState());
+    const range =
+      selectorsForConsole.profile.getPreviewSelectionRange(getState());
+
+    for (const thread of profile.threads) {
+      const { markers } = thread;
+      for (let i = 0; i < markers.length; i++) {
+        const startTime = markers.startTime[i];
+        // Note that Log markers are instant markers, so they only have a start time.
+        if (
+          startTime !== null &&
+          markers.data[i] &&
+          markers.data[i].type === 'Log' &&
+          startTime >= range.start &&
+          startTime <= range.end
+        ) {
+          const data = markers.data[i];
+          const strTimestamp = d2s(
+            profile.meta.startTime + markers.startTime[i]
+          );
+          const processName = thread.processName ?? 'Unknown Process';
+          // TODO: lying about the log level as it's not available yet in the markers
+          const statement = `${strTimestamp} - [${processName} ${thread.pid} ${thread.name}] D/${data.module} ${data.name.trim()}`;
```

Comment:
```suggestion
          const statement = `${strTimestamp} - [${processName} ${thread.pid} ${thread.name}]: D/${data.module} ${data.name.trim()}`;
```

The format for MOZ_LOG has a colon after the ]